src/Avalonia.Base/Controls/ResourceDictionary.cs

```diff
@@ -150,6 +151,21 @@ public void AddDeferred(object key, IDeferredContent deferredContent)
         public void AddNotSharedDeferred(object key, IDeferredContent deferredContent)
             => Add(key, new NotSharedDeferredItem(deferredContent));
 
+        public void AddOrUpdateRange(IEnumerable<KeyValuePair<object, object?>> values)
+        {
+            _suspendNotifications = true;
```

Comment:
Do you still need this flag? Only the `Inner`'s dictionary indexer is called by this method, which shouldn't trigger any notification directly.

Reply:
Pushed the changes just now.